### PR TITLE
CAL-54 Simplify the WKT Geo fields for streaming video metacard

### PR DESCRIPTION
**Pull request by glenhein**
#### What does this PR do?
The frame-center and location metacard attributes for MPEG-TS files/streams
can contain very large amounts of data. This provides a way to simplify the
data by setting a "distance tolerance" that groups near-by data points
together. The distance tolerance is configurable for both the MPEG-TS
input transformer and the MPEG-TS UDP stream monitor.

#### Who is reviewing it?
@rzwiefel @jaymcnallie @jrnorth @kcwire 
#### How should this be tested?
Build and deploy Alliance Experimental. Upload an MPEG-TS w/ KLV file mutlitple times, each time changing the distance tolerance in the mpegts input transformer metatype. Look at the raw metacard with SOLR and note how the overall number of data points change with different values for the distance tolerance. Large tolerance values (~0.01) should result in very few data points. A small tolerance (~0.0001) should result in many data points. You can also ingest the data as a UDP stream.
#### What are the relevant tickets?
CAL-54
#### Checklist:
- [ ] Documentation Updated
- [X] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

File: catalog/video/catalog-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/UdpStreamMonitor.java [hunk 1]
```diff
@@ -331,13 +345,21 @@ public void updateCallback(Map<String, Object> properties) {
                 return;
             }
 
+            if (properties.containsKey(METATYPE_DISTANCE_TOLERANCE) && properties.get(
```
Comment (jrnorth) [inline catalog/video/catalog-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/UdpStreamMonitor.java]: Shouldn't this just be `if (!checkMetaTypeClass(properties, METATYPE_DISTANCE_TOLERANCE, Double.class))`?

Comment (glenhein) [inline catalog/video/catalog-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/UdpStreamMonitor.java]: Normally, yes. But in this case the value can be null, and checkMetaTypeClass indicates an error on null values.

**Comment by jlcsmith** [comment]
@glenhein - please rebase with master to resolve conflicts.  Thanks

**Comment by jlcsmith** [comment]
Built and tested.  LGTM

File: catalog/video/catalog-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/metacard/LineStringMetacardUpdater.java [hunk 2]
```diff
@@ -35,8 +35,25 @@
 
     private final String attributeName;
 
+    private final GeometryFunction geometryFunction;
+
     public LineStringMetacardUpdater(String attributeName) {
         this.attributeName = attributeName;
```
Comment (jrnorth) [inline catalog/video/catalog-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/metacard/LineStringMetacardUpdater.java]: `this(attributeName, GeometryFunction.IDENTITY)`?

File: catalog/video/catalog-mpegts-stream/src/main/resources/OSGI-INF/metatype/metatype.xml [hunk 3]
```diff
@@ -52,6 +52,11 @@
                 name="Metacard Update Initial Delay" id="metacardUpdateInitialDelay" required="false"
                 type="Long" default="2" />
 
+        <AD
+                description="Distance tolerance used to simplify WKT data when merging video chunk metadata with the parent metacard. All vertices in the simplified geometry will be within this distance of the original geometry. The tolerance value must be non-negative."
```
Comment (jrnorth) [inline catalog/video/catalog-mpegts-stream/src/main/resources/OSGI-INF/metatype/metatype.xml]: This description seems a bit complicated. Also, the distance unit should be specified.

Comment (jrnorth) [inline catalog/video/catalog-mpegts-stream/src/main/resources/OSGI-INF/metatype/metatype.xml]: Also there's a typo `numberof` in the `Filename Template` description, could you fix that in this PR?

File: libs/klv/src/test/java/org/codice/alliance/libs/klv/TestGeometryFunction.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+    /**
+     * Since the implementation for the accept method is an empty body, we just call and make sure
+     * none of the visitor methods are called.
+     */
+    @Test
+    public void testAccept() {
+        GeometryFunction.Visitor visitor = mock(GeometryFunction.Visitor.class);
+        GeometryFunction.IDENTITY.accept(visitor);
+        verify(visitor, Mockito.never()).visit(Mockito.any(NormalizeGeometry.class));
```
Comment (jrnorth) [inline libs/klv/src/test/java/org/codice/alliance/libs/klv/TestGeometryFunction.java]: Static import `never()` and `any()`?

File: catalog/video/catalog-mpegts-stream/src/main/resources/OSGI-INF/metatype/metatype.xml (hunk 3 above)
Comment (glenhein) [inline catalog/video/catalog-mpegts-stream/src/main/resources/OSGI-INF/metatype/metatype.xml]: I added the units, but I'm not sure how/if the description should be simplified. @jlcsmith @jaymcnallie

Comment (jlcsmith) [inline catalog/video/catalog-mpegts-stream/src/main/resources/OSGI-INF/metatype/metatype.xml]: Maybe "Distance tolerance used to simplify geospatial metadata during video stream processing.  The tolerance must be non-negative."

File: catalog/video/catalog-mpegts-stream/src/main/resources/OSGI-INF/metatype/metatype.xml [hunk 5]
```diff
@@ -43,7 +43,7 @@
                 type="Long" default="60000"/>
 
         <AD
-                description="Filename template for each chunk. The template may contain any numberof of the sequence '%{date=FORMAT}' where FORMAT is a Java SimpleDateFormat. Must be non-blank."
+                description="Filename template for each chunk. The template may contain any number of of the sequence '%{date=FORMAT}' where FORMAT is a Java SimpleDateFormat. Must be non-blank."
```
Comment (jrnorth) [inline catalog/video/catalog-mpegts-stream/src/main/resources/OSGI-INF/metatype/metatype.xml]: There are two `of`s.

File: catalog/video/catalog-mpegts-stream/src/test/java/org/codice/alliance/video/stream/mpegts/metacard/TestFrameCenterMetacardUpdater.java [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    private FrameCenterMetacardUpdater frameCenterMetacardUpdater;
+
+    @Before
+    public void setup() {
+        frameCenterMetacardUpdater = new FrameCenterMetacardUpdater(GeometryFunction.IDENTITY);
+    }
+
+    @Test
+    public void testToString() {
+        assertThat(frameCenterMetacardUpdater.toString(), notNullValue());
```
Comment (jrnorth) [inline catalog/video/catalog-mpegts-stream/src/test/java/org/codice/alliance/video/stream/mpegts/metacard/TestFrameCenterMetacardUpdater.java]: I'm not sure if there's value in testing the `toString()` method like this. Shouldn't the test be checking whether the returned string is formatted a certain way and contains specific information about the object?

Comment (glenhein) [inline catalog/video/catalog-mpegts-stream/src/test/java/org/codice/alliance/video/stream/mpegts/metacard/TestFrameCenterMetacardUpdater.java]: The only reason I create toString is help the code coverage numbers. I've thought about checking if the string contains at least the class name.

**Comment by jlcsmith** [comment]
needs to be rebased please

**Comment by jaymcnallie** [comment]
👍 - will merge after rebase.

**Pull request merged**